client/extensions/woocommerce/app/settings/email/mailchimp/mailchimp_dashboard.js

```diff
@@ -209,6 +201,9 @@ class MailChimpDashboard extends React.Component {
 				nextProps.successNotice( translate( 'Email settings saved.' ), { duration: 4000 } );
 			}
 		}
+		if ( ( false === this.props.saveSettingsRequest ) && nextProps.saveSettingsRequest ) {
```

Comment:
Is there any chance that a user could click save and click elsewhere in calypso before this triggers? 🤔 wondering if we need these various settings from the tabs to live in the state tree instead of component state.

Reply:
I like the idea about the `protect-form` but this will need to go to a different PR as this would require UX design.